src/native/libs/configure.cmake

```diff
@@ -1092,14 +1148,16 @@ check_symbol_exists(
     sys/sysmacros.h
     HAVE_MAKEDEV_SYSMACROSH)
 
-if (NOT HAVE_MAKEDEV_FILEH AND NOT HAVE_MAKEDEV_SYSMACROSH)
+if (NOT HAVE_MAKEDEV_FILEH AND NOT HAVE_MAKEDEV_SYSMACROSH AND NOT CLR_CMAKE_TARGET_WASI)
   message(FATAL_ERROR "Cannot find the makedev function on this platform.")
 endif()
 
-check_symbol_exists(
-    getgrgid_r
-    grp.h
-    HAVE_GETGRGID_R)
+if (NOT CLR_CMAKE_TARGET_WASI)
```

Comment:
Is this condition necessary because `HAVE_GETGRGID_R` succeeds on wasi and we want to keep it 0 / false?

In that case, we can update its usage in pal_uid.c: `#if HAVE_GETGRGID_R` -> `#if HAVE_GETGRGID_R && !TARGET_WASI` with a comment explaining why, and remove this condition from introspection check.

Reply:
I think what we should do if the symbol is erroneously detected is to override that detection, like we do with HAVE_FORK on L600

```
if (CLR_CMAKE_TARGET_WASI)
  # detection succeeds on wasi but it doesn't actually work
  set(HAVE_GETGRGID_R 0)
endif()
```

Reply:
Yup, that will work too. If we run introspection checks on all platforms uniformly, then in a long run it makes things like preparing cache values list for cross-compilation (https://github.com/dotnet/runtime/blob/5fdc188/eng/native/tryrun.cmake#L29) bit easier; run `build.sh` with `-configureonly` and get flat list of cmake results from `stdout` out of native / reference system and add to the cache list, rather than looking for conditional hierarchies as well.

Reply:
I reverted the change as it doesn't prevent anything at this point. 
Also I'm not sure this was not mistake earlier. 
We will put it back as necessary.